jax/numpy/lax_numpy.py

```diff
@@ -242,9 +242,57 @@ def _promote_args_like(op, *args):
 def _constant_like(x, const):
   return onp.array(const, dtype=_dtype(x))
 
+
+def update_numpydoc(docstr, tabbed, fun, op):
+  original_idx = docstr.find("Original docstring below")
+
+  if tabbed:
+    lines = docstr[original_idx + 25:].split('\n')
+    for idx, line in enumerate(lines):
+      lines[idx] = line.replace('    ', '', 1)
+    docstr = docstr[:original_idx + 25] + '\n'.join(lines)
+
+  try:
+    param_list = [arg for arg in fun.__code__.co_varnames]
+    param_list = [arg for arg in param_list if arg in op.__code__.co_varnames]
+  except AttributeError:
+    begin_idx = docstr.find(fun.__name__ + '(', original_idx) + len(fun.__name__) + 1
+    end_idx = docstr.find(')', begin_idx)
+    param_list = docstr[begin_idx:end_idx].split(', ')
+    for idx, p in enumerate(param_list):
+      eq_idx = p.find('=')
+      if eq_idx == -1:
+        eq_idx = len(p)
+      param = p[:eq_idx]
+      if param[0] == '[':
+        param = param[1:]
+      if param[-1] == ']':
+        param = param[:-1]
+      if param not in op.__code__.co_varnames:
+        param_list[idx] = ''
+    param_list = [param for param in param_list if param != '']
+    parameters = ', '.join(param_list)
+    docstr = docstr[:begin_idx] + parameters + docstr[end_idx:]
+
+  begin_idx = docstr.find("Parameters")
+  begin_idx = docstr.find("--\n", begin_idx) + 2
+  end_idx = docstr.find("Returns", begin_idx)
+
+  parameters = docstr[begin_idx:end_idx]
+  param_list = parameters.replace('\n    ', '@@').split('\n')
+  for idx, p in enumerate(param_list):
+    param = p[:p.find(' : ')].split(", ")[0]
+    if param not in op.__code__.co_varnames:
+      param_list[idx] = ''
+  param_list = [param for param in param_list if param != '']
+  parameters = '\n'.join(param_list).replace('@@', '\n    ')
+  if parameters == "":
+    print(fun.__name__, op.__code__.co_varnames, fun.__code__.co_varnames)
+  return docstr[:begin_idx + 1] + parameters + docstr[end_idx - 2:]
+
 _numpy_signature_re = re.compile(r'^([\w., ]+=)?\s*[\w\.]+\(.*\)$')
 
-def _wraps(fun):
+def _wraps(fun, updates=True, tabbed=True):
   """Like functools.wraps but works with numpy.ufuncs."""
```

Comment:
Can you add a docstring that explains what `updates` and `tabbed` mean? (Basically, if I want to support a new NumPy op, what should I set each of those to based on reading the NumPy docstring?)

Reply:
I wrote an explanation now, is it understandable? (sorry for my bad english)